File: vbox-app/src/main/java/com/kedzie/vbox/api/IDirectory.java
package com.kedzie.vbox.api;

import com.kedzie.vbox.soap.KSOAP;

@KSOAP(prefix="IDirectory")
public interface IDirectory extends IManagedObjectRef {

    @KSOAP(prefix="IDirectory") public String getDirectoryName();

    @KSOAP(prefix="IDirectory") public String getFilter();

    @KSOAP(prefix="IDirectory") public void close();

    @KSOAP(prefix="IDirectory") public String read();
}

File: vbox-app/src/main/java/com/kedzie/vbox/api/IDirectory.kt
package com.kedzie.vbox.api

import android.os.Parcelable
import com.kedzie.vbox.soap.Ksoap
import com.kedzie.vbox.soap.KsoapProxy

@KsoapProxy
@Ksoap(prefix = "IDirectory")
interface IDirectory : IManagedObjectRef {

    suspend fun getDirectoryName():  String
    suspend fun getFilter():  String
    suspend fun close()
    suspend fun read(): String
}